### PR TITLE
SSO OIDC iframe display + repo/code management

### PR DESCRIPTION
**Pull request by atalargo**
- Update version to 0.3.0
- Add iframe display type flow
- Add linters
- Add pre-commit git hook
- Add test environement
- Optimize compile
- Begin add map file for dev
- update gitignore

- [x] Tests:
  - [x] Unit
  - [x] Acceptance [First test \o/ simple page]
- [x] Doc 
- [x] Code

TODO: 
Add more detailed doc on examples

### CODE REVIEW

**Comment by atalargo** [comment]
Jenkins test this please

**Comment by atalargo** [comment]
Jenkins test this please

**Comment by atalargo** [comment]
Jenkins test this please

**Comment by atalargo** [comment]
Jenkins test this please

**Comment by atalargo** [comment]
Jenkins test this please

**Comment by atalargo** [comment]
Jenkins test this please

File: tests/unit/spec/ajax.spec.js [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        onComplete = jasmine.createSpy('onComplete');
+        onSuccess = jasmine.createSpy('onSuccess');
+        onFailure = jasmine.createSpy('onFailure');
+    });
+
+    afterEach(function () {
+        jasmine.Ajax.uninstall();
+    });
+
+    it('should be have an AsmodeeNet.ajax', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: s/be//

File: tests/unit/spec/ajax.spec.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        expect(window.AsmodeeNet).toBeDefined();
+        expect(window.AsmodeeNet.ajax).toBeDefined();
+    });
+
+    it('should can get simple ajax failure GET text/plain query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax failure from a GET text/plain query

File: tests/unit/spec/ajax.spec.js [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        expect(onFailureArgs.length).toBe(3);
+        expect(onFailureArgs[0]).toBeObject();
+        expect(onFailureArgs[0]).toBeInstanceOf('FakeXMLHttpRequest');
+        expect(onFailureArgs[1]).toBeString();
+        expect(onFailureArgs[1]).toEqual('error');
+        expect(onFailureArgs[2]).toBeString();
+        expect(onFailureArgs[2]).toEqual('');
+    });
+
+    it('should can get simple ajax success GET text/plain query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax success from a GET text/plain query

File: tests/unit/spec/ajax.spec.js [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        expect(onSuccessArgs.length).toBe(3);
+        expect(onSuccessArgs[0]).toBeString();
+        expect(onSuccessArgs[0]).toEqual('BOB');
+        expect(onSuccessArgs[1]).toBeString();
+        expect(onSuccessArgs[1]).toEqual('success');
+        expect(onSuccessArgs[2]).toBeObject();
+        expect(onSuccessArgs[2]).toBeInstanceOf('FakeXMLHttpRequest');
+    });
+
+    it('should can get simple ajax failure GET application/json query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax failure from a GET application/json query

File: tests/unit/spec/ajax.spec.js [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+        onFailureArgs = onFailure.calls.mostRecent().args;
+        expect(onFailureArgs.length).toBe(3);
+        expect(onFailureArgs[0]).toBeObject();
+        expect(onFailureArgs[0]).toBeInstanceOf('FakeXMLHttpRequest');
+        expect(onFailureArgs[1]).toBeString();
+        expect(onFailureArgs[1]).toEqual('parsererror');
+        expect(onFailureArgs[2]).toBeString();
+    });
+
+    it('should can get simple ajax success GET application/json query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax success from a GET application/json query

File: tests/unit/spec/ajax.spec.js [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+        expect(onSuccessArgs.length).toBe(3);
+        expect(onSuccessArgs[0]).toBeObject();
+        expect(onSuccessArgs[0]).toEqual({result: ['BOB']});
+        expect(onSuccessArgs[1]).toBeString();
+        expect(onSuccessArgs[1]).toEqual('success');
+        expect(onSuccessArgs[2]).toBeObject();
+        expect(onSuccessArgs[2]).toBeInstanceOf('FakeXMLHttpRequest');
+    });
+
+    it('should can get simple ajax success POST plain/text query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax success from a POST plain/text query

File: tests/unit/spec/ajax.spec.js [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+        expect(onSuccessArgs.length).toBe(3);
+        expect(onSuccessArgs[0]).toBeString();
+        expect(onSuccessArgs[0]).toEqual('OK');
+        expect(onSuccessArgs[1]).toBeString();
+        expect(onSuccessArgs[1]).toEqual('success');
+        expect(onSuccessArgs[2]).toBeObject();
+        expect(onSuccessArgs[2]).toBeInstanceOf('FakeXMLHttpRequest');
+    });
+
+    it('should can get simple ajax success POST application/json query', function () {
```
Comment (laurentbardin) [inline tests/unit/spec/ajax.spec.js]: > should be able to handle a simple ajax success from a POST application/json query

File: .gitignore [hunk 8]
```diff
@@ -1,3 +1,20 @@
...
+tests/unit/reports
+tests/unit/junit
+tests/e2e/reports
+build/an_sso.built.min.js
+build/an_sso.built.min.js.map
+build/an_sso.ext.min.js
+build/an_sso.ext.min.js.map
+build/an_sso.src.cf.js
+build/an_sso.src.cf.js.map
+build/an_sso.src.cf.src.coffee
```
[the remark this answers is not in the document]
Comment (atalargo) [reply .gitignore]: good to know :)

File: README.md [hunk 9]
```diff
@@ -223,7 +228,7 @@ In all situations, the connection data is verified (JWT token verifications), an
 
 Therefore, you **must** call trackCb if you want the whole thing to work when returning from the Identity Server.
 
-If you are looking for a super simple popup example, look at the [examples/index.html](examples/index.html) and [examples/cbpop.html](examples/cbpop.html) files.
+If you are looking for a super simple popup example, look at the [examples/index_popup.html](examples/index.html) and [examples/cbpop.html](examples/cbpop.html) files.
```
[the remark this answers is not in the document]
Comment (atalargo) [reply README.md]: good catch

File: contrib/install_githook.sh [hunk 10]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (atalargo) [reply contrib/install_githook.sh]: arg. no  it's not this one, but another.

File: examples/index_iframe.html [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+            AsmodeeNet.identity({
+                success: function(data) {
+                    echo("My Identity", arguments);
+                }
+            });
+        });
+
+        AsmodeeNet.init({
+            // Required parameters
+            client_id: 'test_direct', //'openid-ffg', //'test_direct', //'openid-dow',
```
[the remark this answers is not in the document]
Comment (atalargo) [reply examples/index_iframe.html]: I leave them for the moment for Francesco...

File: grunt-is/PHPMiddle.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                    res.setHeader('Access-Control-Allow-Headers', 'Authorization, Access-Control-Allow-Origin, Origin, Accept');
+                    if (u.resource() === '/.well-known/openid-configuration') {
+                        // var fname = '/.well-known/openid-configuration';
+                        // if (phpport === 8009) {
+                        //     fname += '.dev';
+                        // } else if (phpport === 8010) {
+                        //     fname += '.test';
+                        // }
+                        // req.url = fname;
+                        // req.originalUrl = fname;
```
[the remark this answers is not in the document]
Comment (atalargo) [reply grunt-is/PHPMiddle.js]: It was the original code fot he same file in the IS...

**Pull request merged**